File: src/models/ssh-key.ts
export interface SSHKey {
	id: string;
	name: string;
	type: "rsa" | "dsa" | "ecdsa" | "ed25519";
	bits?: number; // For RSA keys
	private_key_path: string;
	public_key_path: string;
	passphrase?: string; // Only stored encrypted
	host_ids: string[]; // Hosts using this key
	created_at: Date;
	updated_at: Date;
}

File: src-tauri/src/models/ssh_key.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub enum SSHKeyType {
    #[serde(rename = "rsa")]
    RSA,
    #[serde(rename = "dsa")]
    DSA,
    #[serde(rename = "ecdsa")]
    ECDSA,
    #[serde(rename = "ed25519")]
    Ed25519,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SSHKey {
    pub id: String,
    pub name: String,
    #[serde(rename = "type")]
    pub key_type: SSHKeyType,
    pub bits: Option<u32>,
    pub private_key_path: String,
    pub public_key_path: String,
    pub passphrase: Option<String>,
    pub host_ids: Vec<String>,
    pub created_at: i64,
    pub updated_at: i64,
} 